src/CommandTrait.php

```
@@ -0,0 +1,340 @@
+        }
+      }
+    }
+
+    fclose($pipes[1]);
+    fclose($pipes[2]);
+
+    $this->commandExitCode = proc_close($process);
+    $this->commandDuration = microtime(TRUE) - $started;
+    $this->commandStdout = $stdout;
+    $this->commandStderr = $stderr;
+  }
```

PHPMD (2.15.0)

[warning] The method commandRun() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)